### ISSUE TITLE
UI Shell Header: Take out aria-label from Storybook

### ISSUE THREAD

**Issue by aubrey-oneal** [carbon-design-system/carbon#17029]
Kartarzyna Parjeko brought up in slack the following: In the storybook of UI shell with Switcher, there're aria-labels for links with text. This is redundant, but also a possible accessibility error. I just checked the implementation from my project devs where they added different text in aria-label, creating this error. Can we remove that from the story? [https://github.com/carbon-design-system/carbon/blob/dff81f8c088016a8c5789c284a65bf[…]ages/react/src/components/UIShell/UIShell.HeaderBase.stories.js](https://github.com/carbon-design-system/carbon/blob/dff81f8c088016a8c5789c284a65bfed6e5e3cd8/packages/react/src/components/UIShell/UIShell.HeaderBase.stories.js#L573) 

Gui confirmed this is an [accessibility violation](https://www.ibm.com/able/requirements/requirements/?version=7_1#2_5_3)
